AuthenticationApp/Controller/SignUpVC.swift

```diff
@@ -129,15 +136,41 @@ private extension SignUpVC {
     }
     
     func saveUserData() {
+        saveImage()
         UserDefaults.standard.set(nameTextField.text, forKey: "name")
         UserDefaults.standard.set(emailTextField.text, forKey: "email")
         UserDefaults.standard.set(enterPasswordTextField.text, forKey: "password")
         UserDefaults.standard.set(genderLabel.text, forKey: "gender")
     }
     
+    func saveImage() {
+        guard let data = profileImageView.image?.jpegData(compressionQuality: 0.5) else { return }
+        UserDefaults.standard.set(data, forKey: "profileImage")
+    }
+    
     func goToLogin() {
         if let loginVC = storyboard?.instantiateViewController(identifier: "LoginVC") {
             navigationController?.setViewControllers([loginVC], animated: true)
         }
     }
 }
+
+extension SignUpVC: UINavigationControllerDelegate, UIImagePickerControllerDelegate {
+    
+    func presentImagePicker() {
+        if UIImagePickerController.isSourceTypeAvailable(.savedPhotosAlbum){
+            imagePicker.delegate = self
+            imagePicker.sourceType = .savedPhotosAlbum
+            imagePicker.allowsEditing = false
+            present(imagePicker, animated: true, completion: nil)
+        }
+    }
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        dismiss(animated: true)
+        guard let image = info[.originalImage] as? UIImage else {
```

Comment:
This logic can be simplified using `if let` instead of `guard let`.

Reply:
Done